src/main/kotlin/org/rust/ide/hints/RsInlayParameterHintsProvider.kt

```diff
@@ -23,14 +23,13 @@ import org.rust.stdext.buildList
 
 private val RsPat.inlayInfo: List<InlayInfo>
     get() {
-        val idents = if (this is RsPatIdent) {
-            listOf(this)
-        } else {
-            descendantsOfType<RsPatIdent>()
+        val patBindings = when {
+            this is RsPatIdent -> listOf(patBinding)
+            this is RsPatField && patBinding != null -> listOfNotNull(patBinding)
+            else -> descendantsOfType<RsPatBinding>()
         }
```

Comment:
It seems we can just use `descendantsOfType<RsPatBinding>()` here for all cases

Reply:
Fixed. But we could have done so in the original code too (`descendantsOfType` isn't strict), so I thought it wasn't made for performance reasons.